sdk/search/azure-search/src/main/java/com/azure/search/IndexBatchException.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.exception.AzureException;
+import com.azure.search.models.IndexDocumentsResult;
+import com.azure.search.models.IndexingResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An {@code IndexBatchException} is thrown whenever Azure Cognitive Search index call was only partially successful.
+ * Users can inspect the indexingResults to determine the operation(s) that have failed.
+ */
+public class IndexBatchException extends AzureException {
+    private static final long serialVersionUID = -3478124828996650248L;
+    private static final String MESSAGE_FORMAT = "%s of %s indexing actions in the batch failed. The remaining"
+        + " actions succeeded and modified the index. Check indexingResults for the status of each index action.";
+
+    private final ArrayList<IndexingResult> results;
+
+    /**
+     * Constructs an {@code IndexBatchException} from the given {@link IndexDocumentsResult}.
+     *
+     * @param result The DocumentIndexResult returned from the service.
+     */
+    IndexBatchException(IndexDocumentsResult result) {
```

Comment:
We need to port over `findFailedActionsToRetry` from the .NET SDK. Without it, implementing indexing robustly is a lot more difficult. Adding it later would be a non-breaking change, so we can consider leaving it out for preview and adding it for GA.